os/se/ameba/rtl_security_api_wrapper_tz.c

```diff
@@ -97,6 +89,28 @@ int rtl_ss_flash_read(uint32_t address, uint32_t len, uint8_t *data, int en_disp
 	return ret;
 }
 
+/**
+  * @brief  Verify the Flash Protection Bit is enabled.
+  * @param  none
+  * @return Result.
+  * @retval 1: Flash Protection Bit is enabled.
+  * @retval 0: Flash Protection Bit is disabled.
+  */
+int rtl_verify_flash_protect(void)
+{
+	flash_t flash;
+	unsigned char status_register;
+
+	status_register = 0xFF & flash_get_status(&flash);
+	printf("Status Register: 0x%02x\n", status_register);
+
+	if ((status_register & FLASH_STATUS_BITS) == FLASH_STATUS_BITS) {
+		return 1;
+	} else {
+		return 0;
+	}
+}
```

Comment:
Internal flash size is 8MB, but we protect only first 512KB. Even this situation, can we say that a flash is protected?
Is there anyway where we protect? range?

Reply:
Flash Protection as commented at below:
For Samsung security concern, we only focus on protect the lower 512KB.

This API is to checking whether the flash register bit was enabled.
For protect how much size of flash it was need refer to the flash datasheet.